File: Ambulance.BLL/Models/MedicalCardModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AmbulanceSystem.BLL.Models;
public class MedicalCardModel
{
    public int MedicalCardId { get; set; }
    public int PatientId { get; set; }
    public DateTime? CreationDate { get; set; }
    public int Age { get; set; }
    public string? BloodType { get; set; }
    public decimal? Height { get; set; }
    public decimal? Weight { get; set; }
    public string? Notes { get; set; }
    public List<string> Allergies { get; set; } = new List<string>();
    public List<string> ChronicDiseases { get; set; } = new List<string>();
    public List<MedicalRecordModel> MedicalRecords { get; set; } = new List<MedicalRecordModel>();
}

File: public/Models/MedicalCardModel.js
import { MedicalRecordModel } from "./MedicalRecordModel.js";

export class MedicalCardModel {
    constructor({
        medicalCardId = 0,
        patientId = 0,
        creationDate = null,
        bloodType = null,
        height = null,
        weight = null,
        notes = null,
        allergies = [],
        chronicDiseases = [],
        medicalRecords = []
    } = {}) {
        this.medicalCardId = medicalCardId;
        this.patientId = patientId;
        this.creationDate = creationDate ? new Date(creationDate) : null;
        this.bloodType = bloodType;
        this.height = height;
        this.weight = weight;
        this.notes = notes;
        this.allergies = allergies;
        this.chronicDiseases = chronicDiseases;
        this.medicalRecords = medicalRecords.map(r => new MedicalRecordModel(r));
    }
}